src/core/ext/filters/client_channel/lb_policy/endpoint_list.cc

```diff
@@ -0,0 +1,176 @@
+#include "src/core/lib/config/core_configuration.h"
+#include "src/core/lib/gprpp/debug_location.h"
+#include "src/core/lib/gprpp/orphanable.h"
+#include "src/core/lib/gprpp/ref_counted_ptr.h"
+#include "src/core/lib/iomgr/pollset_set.h"
+#include "src/core/lib/load_balancing/delegating_helper.h"
+#include "src/core/lib/load_balancing/lb_policy.h"
+#include "src/core/lib/load_balancing/lb_policy_registry.h"
+#include "src/core/lib/resolver/server_address.h"
+
+namespace grpc_core {
+
+//
+// EndpointList::Endpoint::Helper
+//
+
+class EndpointList::Endpoint::Helper
+    : public LoadBalancingPolicy::DelegatingChannelControlHelper {
+ public:
+  explicit Helper(RefCountedPtr<Endpoint> endpoint)
+      : endpoint_(std::move(endpoint)) {}
+
+  ~Helper() override { endpoint_.reset(DEBUG_LOCATION, "Helper"); }
+
+  void UpdateState(
+      grpc_connectivity_state state, const absl::Status& status,
+      RefCountedPtr<LoadBalancingPolicy::SubchannelPicker> picker) override {
+    auto old_state = std::exchange(endpoint_->connectivity_state_, state);
+    endpoint_->picker_ = std::move(picker);
+    endpoint_->OnStateUpdate(old_state, state, status);
+  }
+
+ private:
+  LoadBalancingPolicy::ChannelControlHelper* parent_helper() const override {
+    return endpoint_->endpoint_list_->channel_control_helper();
+  }
+
+  RefCountedPtr<Endpoint> endpoint_;
+};
+
+//
+// EndpointList::Endpoint
+//
+
+void EndpointList::Endpoint::Init(
+    const ServerAddress& address, const ChannelArgs& args,
+    std::shared_ptr<WorkSerializer> work_serializer) {
+  ChannelArgs child_args =
+      args.Set(GRPC_ARG_INTERNAL_PICK_FIRST_ENABLE_HEALTH_CHECKING, true)
+          .Set(GRPC_ARG_INTERNAL_PICK_FIRST_OMIT_STATUS_MESSAGE_PREFIX, true);
+  LoadBalancingPolicy::Args lb_policy_args;
+  lb_policy_args.work_serializer = std::move(work_serializer);
+  lb_policy_args.args = child_args;
+  lb_policy_args.channel_control_helper =
+      std::make_unique<Helper>(Ref(DEBUG_LOCATION, "Helper"));
+  child_policy_ =
+      CoreConfiguration::Get().lb_policy_registry().CreateLoadBalancingPolicy(
+          "pick_first", std::move(lb_policy_args));
+  if (GPR_UNLIKELY(endpoint_list_->tracer_ != nullptr)) {
+    gpr_log(GPR_INFO, "[%s %p] endpoint %p: created child policy %p",
+            endpoint_list_->tracer_, endpoint_list_->policy_.get(), this,
+            child_policy_.get());
+  }
+  // Add our interested_parties pollset_set to that of the newly created
+  // child policy. This will make the child policy progress upon activity on
+  // this policy, which in turn is tied to the application's call.
+  grpc_pollset_set_add_pollset_set(
+      child_policy_->interested_parties(),
+      endpoint_list_->policy_->interested_parties());
+  // Update child policy.
+  LoadBalancingPolicy::UpdateArgs update_args;
+  update_args.addresses.emplace().emplace_back(address);
+  update_args.args = child_args;
+  // TODO(roth): If the child reports a non-OK status with the update,
```

Comment:
Should it at least be logged?

Reply:
In this situation, there's no point in logging it, because a non-OK status isn't actually a real failure; it's really just an indication of whether the resolver should consider the update a failure.  We actually have a broader pattern of this issue that occurs in many places, as introduced by #30809.  When we eventually find a good solution for this, we'll fix it everywhere it occurs.

In this particular case, it's actually currently impossible for this to return non-OK anyway, because we are creating one endpoint for each address, and the only case where pick_first will return non-OK is if the address list is empty.  However, that will change when we add support for multiple addresses per endpoint, so I've added a TODO in round_robin.cc to make sure that we properly handle that case.